pkg/odo/cli/component/deploy.go

```diff
@@ -148,6 +152,23 @@ func (do *DeployOptions) Validate() (err error) {
 	}
 	do.ManifestSource = manifestBytes
 
+	// check if manifestSource contains PORT template variable
+	// if it does, then check we have an port setup in env.yaml
+	if bytes.Contains(manifestBytes, []byte("PORT")) {
+		deploymentPort := 0
+		for _, localURL := range do.EnvSpecificInfo.GetURL() {
+			if localURL.Kind == envinfo.ROUTE {
+				deploymentPort = localURL.Port
+				break
+			}
+		}
```

Comment:
should we be using the determine port function here? We could move the function from the adapter to utils and then set the result into `DeployOptions` struct

Reply:
We could do that. Will investigate and update PR if it works.

Reply:
Have moved to pkg/util/util.go